docs/guide/mocking.md

```diff
@@ -1,293 +1,274 @@
+  const currentHour = new Date().getHours()
+  const [open, close] = businessHours
 
-### vi.spyOn
+  if (currentHour > open && currentHour < close) {
+    return { message: 'Success' }
+  }
 
-**Type:** `<T, K extends keyof T>(object: T, method: K, accessType?: 'get' | 'set') => MockInstance`
+  return { message: 'Error' }
+};
 
-Creates a spy on a method or getter/setter of an object.
+describe('purchasing flow', () => {
+  afterEach(() => {
+    // restoring date after each test run
+    vi.restoreCurrentDate()
+  });
 
-```ts
-let apples = 0
-const obj = {
-  getApples: () => 13,
-}
+  it('allows purchases within business hours', () => {
+    // set hour within business hours
+    const date = new Date(2000, 1, 1, 13)
+    vi.mockCurrentDate(date)
 
-const spy = vi.spyOn(obj, 'getApples').mockImplementation(() => apples)
-apples = 1
+    // access Date.now() will result in the date set above
+    expect(purchase()).toEqual({ message: 'Success' })
+  })
 
-expect(obj.getApples()).toBe(1)
+  it('disallows purchases outside of business hours', () => {
+    // set hour outside business hours
+    const date = new Date(2000, 1, 1, 19)
+    vi.mockCurrentDate(date)
 
-expect(spy).toHaveBeenCalled()
-expect(spy).toHaveReturnedWith(1)
+    // access Date.now() will result in the date set above
+    expect(purchase()).toEqual({ message: 'Error' })
+  })
+})
 ```
 
-## Mock methods
-
-### mockName
-
-**Type:** `(name: string) => MockInstance`
-
-Sets internal mock name. Useful to see what mock has failed the assertion.
-
-### getMockName
-
-**Type:** `() => string`
-
-Use it to return the name given to mock with method `.mockName(name)`.
-
-### mockClear
-
-**Type:** `() => MockInstance`
-
-Clears all information about every call. After calling it, [`spy.mock.calls`](#mockcalls), [`spy.mock.returns`](#mockreturns) will return empty arrays. It is useful if you need to clean up spy between different assertions.
-
-If you want this method to be called before each test automatically, you can enable [`clearMocks`](/config/#clearMocks) setting in config.
-
-### mockReset
-
-**Type:** `() => MockInstance`
-
-Does what `mockClear` does and makes inner implementation as an empty function (returning `undefined`, when invoked). This is useful when you want to completely reset a mock back to its initial state.
-
-If you want this method to be called before each test automatically, you can enable [`mockReset`](/config/#mockReset) setting in config.
-
-### mockRestore
-
-**Type:** `() => MockInstance`
-
-Does what `mockRestore` does and restores inner implementation to the original function.
-
-Note that restoring mock from `vi.fn()` will set implementation to an empty function that returns `undefined`. Restoring a `vi.fn(impl)` will restore implementation to `impl`.
+## Functions
+Mock functions (or "spies") observe functions, that are invoked in some other code, allowing you to test its arguments, output or even redeclare its implementation.
```

Comment:
Mocks and spies aren't quite the same; a spy watches and can verify that something has happened (you can spy on `console.log`, for instance), whereas a mock is a fake version of something, often with a spy attached. I'd disconnect the concepts here; focus on spying on existing functions first, then calling a spy function, then mocking functions entirely, and update the example accordingly.

Reply:
Thanks I will make sure the differences between them becomes more clear. I wasn't sure what the most optimal level of handholding would be but what you are saying makes a lot of sense!